src/clean.c

```diff
@@ -189,13 +189,51 @@ static bool is_all_digits(const char *s)
 	return true;
 }
 
+static bool is_all_xdigits(const char *s)
+{
+	for (; *s; s++) {
+		if (!isxdigit(*s)) {
+			return false;
+		}
+	}
+	return true;
+}
+
 static bool is_manifest(const char UNUSED_PARAM *dir, const struct dirent *entry)
 {
 	static const char prefix[] = "Manifest.";
 	static const size_t prefix_len = sizeof(prefix) - 1;
 	return !strncmp(entry->d_name, prefix, prefix_len);
 }
 
+static bool is_hashed_manifest(const char UNUSED_PARAM *dir, const struct dirent *entry)
+{
+	/* check that this has the manifest prefix */
+	if (!is_manifest(dir, entry)) {
+		return false;
+	}
+
+	int i;
+	char *ename, *tmp = NULL;
+	tmp = ename = strdup_or_die(entry->d_name);
+
+	/* check for the correct number of '.' characters
+	 * (i.e. Manifest.bundlename.hashvalue). Note, this will not clean up
+	 * iterative manifests (Manifest.bundlename.I) because I is not a valid
```

Comment:
I would check the hash size too. This isn't a problem for D manifests because of the number of '.'s, but we got close.

Reply:
As you said the delta manifests will fail this check because of the mismatch of '.'s so I don't see the point of adding another check there. Also, my iterative manifest comment is incorrect since the format for that is `Manifest.bundlename.I.version` so it will also fail this test. I'll update that comment.